Not compiling on latest version

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
Running on WSL with PIP

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->

**ESPHome version (latest production, beta, dev branch)**
<!--
ESPHome version like v1.14 or 1.15-dev
-->
```
dclobato@DESK-DANIEL:~/ESPhome/HA$ esphome version
Version: 1.14.5
dclobato@DESK-DANIEL:~/ESPhome/HA$
```

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->


**Description of problem:**
For some reason, esphome is not compiling anymore.  I updated platformio to latest version, and a lot of bugs emerged on configurations that were working fine

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: ${nomeDoNodo}
  platform: ESP8266
  board: esp8285
  build_path: binarios/${nomeDoNodo}

<<: !include common/common.yaml
<<: !include common/status_led.yaml

sensor:
  - !include sensor/wifi-signal.yaml
  - !include sensor/uptime.yaml

switch:
  - !include switch/restart.yaml

# Enable logging
logger:

binary_sensor:
  - !include binary_sensor/status.yaml
  - platform: gpio
    pin: GPIO00
    id: reset
    internal: true
    filters:
      - invert:
      - delayed_off: 10ms
    on_press:
      - light.toggle:
          id: light_1

  - platform: gpio
    name: "Botão ${dePrep} ${Nome}"
    pin: GPIO04
    id: switch_1
    on_press:
      then:
        - light.toggle:
            id: light_1
    on_release:
      then:
        - light.toggle:
            id: light_1

output:
  - platform: gpio
    pin: GPIO12
    id: relay_1

light:
  - platform: binary
    name: "Lâmpada ${emPrep} ${Nome}"
    id: light_1
    output: relay_1
```

**Logs (if applicable):**
<!--
Please copy the debug log here. If possible, also connect to the ESP over USB and copy those logs into the backticks.
-->
```
dclobato@DESK-DANIEL:~$ sudo pip3 install esphome
Collecting esphome
  Using cached esphome-1.14.5-py2.py3-none-any.whl (1.4 MB)
Requirement already satisfied: paho-mqtt==1.5.0 in /usr/local/lib/python3.8/dist-packages (from esphome) (1.5.0)
Requirement already satisfied: protobuf==3.11.3 in /usr/local/lib/python3.8/dist-packages (from esphome) (3.11.3)
Requirement already satisfied: pyserial==3.4 in /usr/lib/python3/dist-packages (from esphome) (3.4)
Requirement already satisfied: tornado==5.1.1 in /usr/local/lib/python3.8/dist-packages (from esphome) (5.1.1)
Requirement already satisfied: voluptuous==0.11.7 in /usr/local/lib/python3.8/dist-packages (from esphome) (0.11.7)
Requirement already satisfied: ifaddr==0.1.6 in /usr/local/lib/python3.8/dist-packages (from esphome) (0.1.6)
Requirement already satisfied: colorlog==4.1.0 in /usr/local/lib/python3.8/dist-packages (from esphome) (4.1.0)
Processing /root/.cache/pip/wheels/22/76/4e/8451b93142686519520b707538b8ced937c1b7a3e79f369ef5/platformio-4.3.4-py3-none-any.whl
Requirement already satisfied: pytz==2020.1 in /usr/local/lib/python3.8/dist-packages (from esphome) (2020.1)
Requirement already satisfied: esptool==2.8 in /usr/local/lib/python3.8/dist-packages (from esphome) (2.8)
Requirement already satisfied: PyYAML==5.3.1 in /usr/lib/python3/dist-packages (from esphome) (5.3.1)
Requirement already satisfied: tzlocal==2.0.0 in /usr/local/lib/python3.8/dist-packages (from esphome) (2.0.0)
Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (from protobuf==3.11.3->esphome) (45.2.0)
Requirement already satisfied: six>=1.9 in /usr/lib/python3/dist-packages (from protobuf==3.11.3->esphome) (1.14.0)
Requirement already satisfied: click<8,>=5 in /usr/lib/python3/dist-packages (from platformio==4.3.4->esphome) (7.0)
Requirement already satisfied: tabulate<1,>=0.8.3 in /usr/local/lib/python3.8/dist-packages (from platformio==4.3.4->esphome) (0.8.7)
Requirement already satisfied: requests<3,>=2.4.0 in /usr/lib/python3/dist-packages (from platformio==4.3.4->esphome) (2.22.0)
Requirement already satisfied: semantic-version<3,>=2.8.1 in /usr/local/lib/python3.8/dist-packages (from platformio==4.3.4->esphome) (2.8.5)
Requirement already satisfied: bottle<0.13 in /usr/local/lib/python3.8/dist-packages (from platformio==4.3.4->esphome) (0.12.18)
Requirement already satisfied: marshmallow>=2 in /usr/local/lib/python3.8/dist-packages (from platformio==4.3.4->esphome) (3.7.0)
Requirement already satisfied: pyelftools<1,>=0.25 in /usr/local/lib/python3.8/dist-packages (from platformio==4.3.4->esphome) (0.26)
Requirement already satisfied: colorama in /usr/lib/python3/dist-packages (from platformio==4.3.4->esphome) (0.4.3)
Requirement already satisfied: pyaes in /usr/local/lib/python3.8/dist-packages (from esptool==2.8->esphome) (1.6.1)
Requirement already satisfied: ecdsa in /usr/local/lib/python3.8/dist-packages (from esptool==2.8->esphome) (0.15)
Installing collected packages: platformio, esphome
Successfully installed esphome-1.14.5 platformio-4.3.4
WARNING: You are using pip version 20.1.1; however, version 20.2.1 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
dclobato@DESK-DANIEL:~$ sudo /usr/bin/python3 -m pip install --upgrade pip
Collecting pip
  Downloading pip-20.2.1-py2.py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 3.0 MB/s
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.1.1
    Uninstalling pip-20.1.1:
      Successfully uninstalled pip-20.1.1
Successfully installed pip-20.2.1
dclobato@DESK-DANIEL:~$ cd ESPhome/
dclobato@DESK-DANIEL:~/ESPhome$ cd HA
dclobato@DESK-DANIEL:~/ESPhome/HA$ esphome luz-do-corredor.yaml compile
INFO Reading configuration luz-do-corredor.yaml...
INFO Detected timezone 'TZ' with UTC offset -3
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d binarios/luzcorredor
******************************************************************************************************************************
If you like PlatformIO, please:
- follow us on Twitter to stay up-to-date on the latest project news > https://twitter.com/PlatformIO_Org
- star it on GitHub > https://github.com/platformio/platformio
- try PlatformIO IDE for embedded development > https://platformio.org/platformio-ide
******************************************************************************************************************************

Processing luzcorredor (board: esp8285; framework: arduino; platform: espressif8266@2.2.3)
------------------------------------------------------------------------------------------------------------------------------
PlatformManager: Installing espressif8266 @ 2.2.3
espressif8266 @ 2.2.3 has been successfully installed!
The platform 'espressif8266@2.2.3' has been successfully installed!
The rest of packages will be installed automatically depending on your build environment.
PackageManager: Installing toolchain-xtensa @ <3
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
toolchain-xtensa @ 2.100100.200706 has been successfully installed!
PackageManager: Installing framework-arduinoespressif8266 @ ~2.20502.0
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
framework-arduinoespressif8266 @ 2.20502.0 has been successfully installed!
PackageManager: Installing tool-esptool @ <2
tool-esptool @ 1.413.0 has been successfully installed!
PackageManager: Installing tool-esptoolpy @ ~1.20600.0
tool-esptoolpy @ 1.20600.0 has been successfully installed!
CorePackageManager: Installing tool-scons @ ~3.30102.0
Downloading  [####################################]  100%
tool-scons @ 3.30102.0 has been successfully installed!
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
PACKAGES:
 - framework-arduinoespressif8266 2.20502.0 (2.5.2)
 - tool-esptool 1.413.0 (4.13)
 - tool-esptoolpy 1.20600.0 (2.6.0)
 - toolchain-xtensa 2.100100.200706 (10.1.0)
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.2
|   |-- <ESP8266WiFi> 1.0
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncWebServer-esphome> 1.2.6
|   |-- <ESPAsyncTCP-esphome> 1.2.2
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|-- <DNSServer> 1.1.1
|   |-- <ESP8266WiFi> 1.0
Compiling .pioenvs/luzcorredor/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/luzcorredor/src/esphome/components/api/user_services.cpp.o
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/user_services.h:3,
                 from src/esphome/components/api/user_services.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/proto.h:3,
                 from src/esphome/components/api/api_pb2.h:3,
                 from src/esphome/components/api/api_pb2.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/proto.h:3,
                 from src/esphome/components/api/api_pb2.h:3,
                 from src/esphome/components/api/api_pb2_service.h:3,
                 from src/esphome/components/api/api_pb2_service.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/proto.h:3,
                 from src/esphome/components/api/proto.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/subscribe_state.h:3,
                 from src/esphome/components/api/subscribe_state.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/api_connection.h:3,
                 from src/esphome/components/api/api_connection.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/list_entities.h:3,
                 from src/esphome/components/api/list_entities.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src/esphome/components/api/api_server.h:3,
                 from src/esphome/components/api/api_server.cpp:1:
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
/home/dclobato/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
*** [.pioenvs/luzcorredor/src/esphome/components/api/user_services.cpp.o] Error 1
*** [.pioenvs/luzcorredor/src/esphome/components/api/api_pb2_service.cpp.o] Error 1
*** [.pioenvs/luzcorredor/src/esphome/components/api/proto.cpp.o] Error 1
In file included from src/esphome/components/api/api_server.h:12,
                 from src/esphome/components/api/subscribe_state.h:47,
                 from src/esphome/components/api/subscribe_state.cpp:1:
src/esphome/components/api/homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src/esphome/components/api/homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src/esphome/components/api/subscribe_state.h:6,
                 from src/esphome/components/api/subscribe_state.cpp:1:
src/esphome/components/api/util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
In file included from src/esphome/components/api/api_server.h:12,
                 from src/esphome/components/api/list_entities.h:52,
                 from src/esphome/components/api/list_entities.cpp:1:
src/esphome/components/api/homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src/esphome/components/api/homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src/esphome/components/api/list_entities.h:5,
                 from src/esphome/components/api/list_entities.cpp:1:
src/esphome/components/api/util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
In file included from src/esphome/components/api/api_server.h:12,
                 from src/esphome/components/api/api_server.cpp:1:
src/esphome/components/api/homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src/esphome/components/api/homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src/esphome/components/api/api_server.h:9,
                 from src/esphome/components/api/api_server.cpp:1:
src/esphome/components/api/util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
In file included from src/esphome/components/api/api_server.h:12,
                 from src/esphome/components/api/api_connection.h:7,
                 from src/esphome/components/api/api_connection.cpp:1:
src/esphome/components/api/homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src/esphome/components/api/homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src/esphome/components/api/api_server.h:9,
                 from src/esphome/components/api/api_connection.h:7,
                 from src/esphome/components/api/api_connection.cpp:1:
src/esphome/components/api/util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
*** [.pioenvs/luzcorredor/src/esphome/components/api/api_pb2.cpp.o] Error 1
*** [.pioenvs/luzcorredor/src/esphome/components/api/subscribe_state.cpp.o] Error 1
*** [.pioenvs/luzcorredor/src/esphome/components/api/list_entities.cpp.o] Error 1
*** [.pioenvs/luzcorredor/src/esphome/components/api/api_server.cpp.o] Error 1
*** [.pioenvs/luzcorredor/src/esphome/components/api/api_connection.cpp.o] Error 1
================================================ [FAILED] Took 61.04 seconds ================================================
```

**Additional information and things you've tried:**



<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

The same happens on fresh install of esphome on Windows 10

```
C:\Windows\system32>pip3 install esphome
Collecting esphome
  Using cached esphome-1.14.5-py2.py3-none-any.whl (1.4 MB)
Collecting colorlog==4.1.0
  Using cached colorlog-4.1.0-py2.py3-none-any.whl (14 kB)
Collecting tornado==5.1.1
  Using cached tornado-5.1.1.tar.gz (516 kB)
Collecting tzlocal==2.0.0
  Using cached tzlocal-2.0.0-py2.py3-none-any.whl (15 kB)
Collecting voluptuous==0.11.7
  Using cached voluptuous-0.11.7.tar.gz (45 kB)
Collecting protobuf==3.11.3
  Using cached protobuf-3.11.3-py2.py3-none-any.whl (434 kB)
Collecting esptool==2.8
  Using cached esptool-2.8.tar.gz (84 kB)
Collecting ifaddr==0.1.6
  Using cached ifaddr-0.1.6.tar.gz (6.5 kB)
Collecting pyserial==3.4
  Using cached pyserial-3.4-py2.py3-none-any.whl (193 kB)
Collecting paho-mqtt==1.5.0
  Using cached paho-mqtt-1.5.0.tar.gz (99 kB)
Collecting pytz==2020.1
  Using cached pytz-2020.1-py2.py3-none-any.whl (510 kB)
Collecting platformio==4.3.4
  Using cached platformio-4.3.4.tar.gz (190 kB)
Collecting PyYAML==5.3.1
  Using cached PyYAML-5.3.1-cp38-cp38-win_amd64.whl (219 kB)
Collecting colorama; sys_platform == "win32"
  Using cached colorama-0.4.3-py2.py3-none-any.whl (15 kB)
Requirement already satisfied: setuptools in c:\program files\python38\lib\site-packages (from protobuf==3.11.3->esphome) (41.2.0)
Requirement already satisfied: six>=1.9 in c:\program files\python38\lib\site-packages (from protobuf==3.11.3->esphome) (1.15.0)
Collecting pyaes
  Using cached pyaes-1.6.1.tar.gz (28 kB)
Collecting ecdsa
  Using cached ecdsa-0.15-py2.py3-none-any.whl (100 kB)
Collecting bottle<0.13
  Using cached bottle-0.12.18-py3-none-any.whl (89 kB)
Collecting click!=7.1,!=7.1.1,<8,>=5
  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)
Requirement already satisfied: requests<3,>=2.4.0 in c:\program files\python38\lib\site-packages (from platformio==4.3.4->esphome) (2.24.0)
Collecting semantic_version<3,>=2.8.1
  Using cached semantic_version-2.8.5-py2.py3-none-any.whl (15 kB)
Collecting tabulate<1,>=0.8.3
  Using cached tabulate-0.8.7-py3-none-any.whl (24 kB)
Collecting pyelftools<1,>=0.25
  Using cached pyelftools-0.26-py2.py3-none-any.whl (136 kB)
Collecting marshmallow>=2
  Downloading marshmallow-3.7.1-py2.py3-none-any.whl (45 kB)
     |████████████████████████████████| 45 kB 3.4 MB/s
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\program files\python38\lib\site-packages (from requests<3,>=2.4.0->platformio==4.3.4->esphome) (1.25.10)
Requirement already satisfied: chardet<4,>=3.0.2 in c:\program files\python38\lib\site-packages (from requests<3,>=2.4.0->platformio==4.3.4->esphome) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in c:\program files\python38\lib\site-packages (from requests<3,>=2.4.0->platformio==4.3.4->esphome) (2020.6.20)
Requirement already satisfied: idna<3,>=2.5 in c:\program files\python38\lib\site-packages (from requests<3,>=2.4.0->platformio==4.3.4->esphome) (2.10)
Using legacy setup.py install for tornado, since package 'wheel' is not installed.
Using legacy setup.py install for voluptuous, since package 'wheel' is not installed.
Using legacy setup.py install for esptool, since package 'wheel' is not installed.
Using legacy setup.py install for ifaddr, since package 'wheel' is not installed.
Using legacy setup.py install for paho-mqtt, since package 'wheel' is not installed.
Using legacy setup.py install for platformio, since package 'wheel' is not installed.
Using legacy setup.py install for pyaes, since package 'wheel' is not installed.
Installing collected packages: colorama, colorlog, tornado, pytz, tzlocal, voluptuous, protobuf, pyserial, pyaes, ecdsa, esptool, ifaddr, paho-mqtt, bottle, click, semantic-version, tabulate, pyelftools, marshmallow, platformio, PyYAML, esphome
    Running setup.py install for tornado ... done
    Running setup.py install for voluptuous ... done
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.12.2
    Uninstalling protobuf-3.12.2:
      Successfully uninstalled protobuf-3.12.2
    Running setup.py install for pyaes ... done
    Running setup.py install for esptool ... done
    Running setup.py install for ifaddr ... done
    Running setup.py install for paho-mqtt ... done
    Running setup.py install for platformio ... done
Successfully installed PyYAML-5.3.1 bottle-0.12.18 click-7.1.2 colorama-0.4.3 colorlog-4.1.0 ecdsa-0.15 esphome-1.14.5 esptool-2.8 ifaddr-0.1.6 marshmallow-3.7.1 paho-mqtt-1.5.0 platformio-4.3.4 protobuf-3.11.3 pyaes-1.6.1 pyelftools-0.26 pyserial-3.4 pytz-2020.1 semantic-version-2.8.5 tabulate-0.8.7 tornado-5.1.1 tzlocal-2.0.0 voluptuous-0.11.7
WARNING: You are using pip version 20.1.1; however, version 20.2.1 is available.
You should consider upgrading via the 'c:\program files\python38\python.exe -m pip install --upgrade pip' command.
```

Now, trying to compile the same file that was compiling before updating platformio.

```
C:\>esphome luz-do-corredor.yaml compile
←[32mINFO Reading configuration luz-do-corredor.yaml...←[0m
←[32mINFO Detected timezone 'TZ' with UTC offset -3←[0m
←[32mINFO Generating C++ source...←[0m
←[32mINFO Core config or version changed, cleaning build files...←[0m
←[32mINFO Compiling app...←[0m
←[32mINFO Running:  platformio run -d binarios/luzcorredor←[0m
************************************************************************************************************************
If you like PlatformIO, please:
- follow us on Twitter to stay up-to-date on the latest project news > https://twitter.com/PlatformIO_Org
- star it on GitHub > https://github.com/platformio/platformio
- try PlatformIO IDE for embedded development > https://platformio.org/platformio-ide
************************************************************************************************************************

Processing luzcorredor (board: esp8285; framework: arduino; platform: espressif8266@2.2.3)
------------------------------------------------------------------------------------------------------------------------
PlatformManager: Installing espressif8266 @ 2.2.3
espressif8266 @ 2.2.3 has been successfully installed!
The platform 'espressif8266@2.2.3' has been successfully installed!
The rest of packages will be installed automatically depending on your build environment.
PackageManager: Installing toolchain-xtensa @ <3
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
toolchain-xtensa @ 2.100100.200706 has been successfully installed!
PackageManager: Installing framework-arduinoespressif8266 @ ~2.20502.0
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
framework-arduinoespressif8266 @ 2.20502.0 has been successfully installed!
PackageManager: Installing tool-esptool @ <2
tool-esptool @ 1.413.0 has been successfully installed!
PackageManager: Installing tool-esptoolpy @ ~1.20600.0
tool-esptoolpy @ 1.20600.0 has been successfully installed!
CorePackageManager: Installing tool-scons @ ~3.30102.0
Downloading  [####################################]  100%
tool-scons @ 3.30102.0 has been successfully installed!
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/esp8285.html
PLATFORM: Espressif 8266 2.2.3 > Generic ESP8285 Module
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
PACKAGES:
 - framework-arduinoespressif8266 2.20502.0 (2.5.2)
 - tool-esptool 1.413.0 (4.13)
 - tool-esptoolpy 1.20600.0 (2.6.0)
 - toolchain-xtensa 2.100100.200706 (10.1.0)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Looking for ESPAsyncTCP-esphome library in registry
Found: https://platformio.org/lib/show/6757/ESPAsyncTCP-esphome
LibraryManager: Installing id=6757 @ 1.2.2
ESPAsyncTCP-esphome @ 1.2.2 has been successfully installed!
Looking for ESPAsyncWebServer-esphome library in registry
Found: https://platformio.org/lib/show/6758/ESPAsyncWebServer-esphome
LibraryManager: Installing id=6758 @ 1.2.6

Downloading  [############################--------]   78%
Downloading  [#############################-------]   81%
Downloading  [##############################------]   84%
Downloading  [###############################-----]   87%
Downloading  [################################----]   90%
Downloading  [#################################---]   93%
Downloading  [##################################--]   96%
Downloading  [####################################]  100%
ESPAsyncWebServer-esphome @ 1.2.6 has been successfully installed!
Installing dependencies
Looking for ESPAsyncTCP-esphome library in registry
Found: https://platformio.org/lib/show/6757/ESPAsyncTCP-esphome
LibraryManager: Installing id=6757
ESPAsyncTCP-esphome @ 1.2.3 has been successfully installed!
Looking for AsyncTCP-esphome library in registry
Found: https://platformio.org/lib/show/6798/AsyncTCP-esphome
LibraryManager: Installing id=6798
AsyncTCP-esphome @ 1.1.1 has been successfully installed!
Looking for Hash library in registry
Warning! Library `{'name': 'Hash', 'platforms': ['espressif8266']}` has not been found in PlatformIO Registry.
You can ignore this message, if `{'name': 'Hash', 'platforms': ['espressif8266']}` is a built-in library (included in framework, SDK). E.g., SPI, Wire, etc.
Looking for Update library in registry
Warning! Library `{'name': 'Update', 'requirements': None}` has not been found in PlatformIO Registry.
You can ignore this message, if `{'name': 'Update', 'requirements': None}` is a built-in library (included in framework, SDK). E.g., SPI, Wire, etc.
Found 31 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.2
|   |-- <ESP8266WiFi> 1.0
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncWebServer-esphome> 1.2.6
|   |-- <ESPAsyncTCP-esphome> 1.2.2
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|-- <DNSServer> 1.1.1
|   |-- <ESP8266WiFi> 1.0
Building in release mode
Compiling .pioenvs\luzcorredor\src\esphome\components\api\api_connection.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\api_pb2.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\api_pb2_service.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\api_server.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\list_entities.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\proto.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\subscribe_state.cpp.o
Compiling .pioenvs\luzcorredor\src\esphome\components\api\user_services.cpp.o
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\proto.h:3,
                 from src\esphome\components\api\api_pb2.h:3,
                 from src\esphome\components\api\api_pb2.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\user_services.h:3,
                 from src\esphome\components\api\user_services.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\proto.h:3,
                 from src\esphome\components\api\api_pb2.h:3,
                 from src\esphome\components\api\api_pb2_service.h:3,
                 from src\esphome\components\api\api_pb2_service.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\proto.h:3,
                 from src\esphome\components\api\proto.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\list_entities.h:3,
                 from src\esphome\components\api\list_entities.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\api_server.h:3,
                 from src\esphome\components\api\api_server.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\subscribe_state.h:3,
                 from src\esphome\components\api\subscribe_state.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src/esphome/core/component.h:5,
                 from src\esphome\components\api\api_connection.h:3,
                 from src\esphome\components\api\api_connection.cpp:1:
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:270:12: error: 'isinf' has not been declared in 'std'
  270 | using std::isinf;
      |            ^~~~~
C:\Users\danie\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:271:12: error: 'isnan' has not been declared in 'std'
  271 | using std::isnan;
      |            ^~~~~
In file included from src\esphome\components\api\api_server.h:12,
                 from src\esphome\components\api\list_entities.h:52,
                 from src\esphome\components\api\list_entities.cpp:1:
src\esphome\components\api\homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src\esphome\components\api\homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src\esphome\components\api\list_entities.h:5,
                 from src\esphome\components\api\list_entities.cpp:1:
src\esphome\components\api\util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
*** [.pioenvs\luzcorredor\src\esphome\components\api\api_pb2_service.cpp.o] Error 1
*** [.pioenvs\luzcorredor\src\esphome\components\api\user_services.cpp.o] Error 1
In file included from src\esphome\components\api\api_server.h:12,
                 from src\esphome\components\api\subscribe_state.h:47,
                 from src\esphome\components\api\subscribe_state.cpp:1:
src\esphome\components\api\homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src\esphome\components\api\homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src\esphome\components\api\subscribe_state.h:6,
                 from src\esphome\components\api\subscribe_state.cpp:1:
src\esphome\components\api\util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
In file included from src\esphome\components\api\api_server.h:12,
                 from src\esphome\components\api\api_server.cpp:1:
src\esphome\components\api\homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src\esphome\components\api\homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src\esphome\components\api\api_server.h:9,
                 from src\esphome\components\api\api_server.cpp:1:
src\esphome\components\api\util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
*** [.pioenvs\luzcorredor\src\esphome\components\api\proto.cpp.o] Error 1
In file included from src\esphome\components\api\api_server.h:12,
                 from src\esphome\components\api\api_connection.h:7,
                 from src\esphome\components\api\api_connection.cpp:1:
src\esphome\components\api\homeassistant_service.h: In member function 'void esphome::api::HomeAssistantServiceCallAction<Ts>::play(Ts ...)':
src\esphome\components\api\homeassistant_service.h:54:18: warning: invalid use of incomplete type 'class esphome::api::APIServer'
   54 |     this->parent_->send_homeassistant_service_call(resp);
      |                  ^~
In file included from src\esphome\components\api\api_server.h:9,
                 from src\esphome\components\api\api_connection.h:7,
                 from src\esphome\components\api\api_connection.cpp:1:
src\esphome\components\api\util.h:13:7: note: forward declaration of 'class esphome::api::APIServer'
   13 | class APIServer;
      |       ^~~~~~~~~
*** [.pioenvs\luzcorredor\src\esphome\components\api\list_entities.cpp.o] Error 1
*** [.pioenvs\luzcorredor\src\esphome\components\api\api_pb2.cpp.o] Error 1
*** [.pioenvs\luzcorredor\src\esphome\components\api\subscribe_state.cpp.o] Error 1
*** [.pioenvs\luzcorredor\src\esphome\components\api\api_server.cpp.o] Error 1
*** [.pioenvs\luzcorredor\src\esphome\components\api\api_connection.cpp.o] Error 1
=============================================================== [FAILED] Took 72.66 seconds ===============================================================
```

--- a/pkgs/tools/misc/esphome/default.nix
+++ b/pkgs/tools/misc/esphome/default.nix
@@ -1,21 +1,21 @@
-{ lib, python3, platformio, esptool, git, protobuf3_11, fetchpatch }:
+{ lib, python3, platformio, esptool, git, protobuf3_12, fetchpatch }:
 
 let
   python = python3.override {
     packageOverrides = self: super: {
       protobuf = super.protobuf.override {
-        protobuf = protobuf3_11;
+        protobuf = protobuf3_12;
       };
     };
   };
 
 in python.pkgs.buildPythonApplication rec {
   pname = "esphome";
-  version = "1.14.5";
+  version = "1.15.0";
 
   src = python.pkgs.fetchPypi {
     inherit pname version;
-    sha256 = "176mi361677d5cqbi0hn52kky845byjs6gdad8pdhihyjgv7a9y9";
+    sha256 = "1npw5rm0mcfydrxxyk4bgzj5vmcwasw1y2idc3bw0ksgv9q69ird";
   };
 
   ESPHOME_USE_SUBPROCESS = "";
@@ -28,7 +28,7 @@ in python.pkgs.buildPythonApplication rec {
 
   # remove all version pinning (E.g tornado==5.1.1 -> tornado)
   postPatch = ''
-    sed -i -e "s/==[0-9.]*//" setup.py
+    sed -i -e "s/==[0-9.]*//" requirements.txt
   '';
 
   makeWrapperArgs = [

--- a/pkgs/tools/misc/esphome/default.nix
+++ b/pkgs/tools/misc/esphome/default.nix
@@ -11,11 +11,11 @@ let
 
 in python.pkgs.buildPythonApplication rec {
   pname = "esphome";
-  version = "1.15.0";
+  version = "1.15.2";
 
   src = python.pkgs.fetchPypi {
     inherit pname version;
-    sha256 = "1npw5rm0mcfydrxxyk4bgzj5vmcwasw1y2idc3bw0ksgv9q69ird";
+    sha256 = "1wnmgn0q4n2vp2cdwsc36acsy7c7w5vyxdglii3432mr5drrgcsx";
   };
 
   ESPHOME_USE_SUBPROCESS = "";

